### ISSUE TITLE
Catchup: Flussonic timeshift issues

### ISSUE THREAD

**Issue by SerhiyRomanov** [kodi-pvr/pvr.iptvsimple#392]
I use playlist like

MPEG-TS:
```
#EXTINF:-1 timeshift="7" catchup-days="7" catchup-type="flussonic-ts" tvg-id="amedia-premium"  group-title="Кино",Amedia Premium HD
http://str04.ottg.tv/3610/mpegts?username=login&password=pass&token=token&ch_id=251&req_host=pkjX3BL
```
HLS:
```
#EXTINF:-1 timeshift="7" catchup-days="7" catchup-type="flussonic" tvg-id="amedia-premium"  group-title="Кино",Amedia Premium HD
http://str04.ottg.tv/3610/video.m3u8?username=login&password=pass&token=token&ch_id=251&req_host=pkjX3BL
```

For HLS - catchup works, but small timeshift (or forward up/back) for 3, 5, 10 seconds are not even available. 

For TS all this stuff works correctly but the EPG Catchup URL looks like `http://str04.ottg.tv/3610/mpegts?username=login&password=pass&token=token&ch_id=251&req_host=pkjX3BL&utc=1590733440&lutc=1590759254 `
It's weird that its works with Flussonic, I have no idea why.

Kodi (19.0-ALPHA1 Git:20200529-376cca988c). Platform: Linux x86 64-bit
pvr.iptvsimple v5.1.0 installed
inputstream.adaptive v2.6.0 installed

**Comment by phunkyfish** [comment]
Did 3,5 and 10 seconds used to work for HLS?

Also can you tell me what is wrong with the EPG catchup url?

**Comment by SerhiyRomanov** [comment]
> Did 3,5 and 10 seconds used to work for HLS?

Yes, it worked later. Also, it doesn't work for other HLS playlists now, like this one
```
#EXTINF:0 catchup="fs" catchup-days="3" tvg-name="1+1" group-title="UA Украина",1+1
http://am01.spr24.net/151/index.m3u8?token=token
```
Maybe the problem is with `catchup-type="flussonic"`? You introduced this tag recently. Does addon detect such stream as Flussonic HLS correctly?

> Also can you tell me what is wrong with the EPG catchup URL?

Previously it generated catchup like 
http://am01.spr24.net/151/timeshift_abs-1590755100.ts?token=token

Now it appends `&utc=1590733440&lutc=1590759254`, but it also works, but I'm not sure that it will work for all IPTV servers.
Also, does addon detect such stream as Flussonic TS correctly?

**Comment by phunkyfish** [comment]
Ok, let me look at this next week.

**Comment by phunkyfish** [comment]
Is this issue only effecting flussonic? Also do you know in which version this issue began?

**Comment by SerhiyRomanov** [comment]
I've tested with Shift (SIPTV) and it's work correctly for it. So, I guess, it is the only a problem with Flussonic.
Again, I guess, that this bug appeared in 5.1.0, after you implemented logic for catchup-type tag in scope of #388

**Comment by phunkyfish** [comment]
Ok, cool. Thanks for checking.

**Comment by phunkyfish** [comment]
Are you saying that 3, 5 and 10 second seek works for TS but not for HLS?

**Comment by phunkyfish** [comment]
This PR includes the wrong URL fix: https://github.com/kodi-pvr/pvr.iptvsimple/pull/395/files

I think the HLS seek is related to a continuation feature added for xc streams ffmpegdirect.

**Comment by phunkyfish** [comment]
Do you have a debug log for the HLS streams where 3,5 and 10 second seeks don't occur?

**Comment by phunkyfish** [comment]
Ok, I added a commit to #395. That might fix the 3,5,10 issues.

Let me know.

**Comment by SerhiyRomanov** [comment]
@phunkyfish I've tested 5.2.0 version of addon - the formatting URL issue is resolved.

But the timeshift for archived programmes is not available now for both HLS and TS. 
As I see, the new section Timeshift appeared in Settings, I've enabled both checkboxes here. Should I do anything else to get it worked?

//to be sure - I mean the ability to rewind the archived programme when I say about "Timeshift", is it okay?
